### PR TITLE
fixed ONNX softmax support documentation

### PR DESCRIPTION
**Pull request by dillonalaird**
When exporting PyTorch models to ONNX, softmax is not fully supported:

https://github.com/pytorch/pytorch/pull/4592/commits/2ad535974d35d2a014320401e54c7ee9a653e3d9

https://github.com/onnx/onnx-caffe2/blob/master/tests/ONNXOpCoverage.md

Updated ONNX docs stating that softmax only works when dim=-1.

### CODE REVIEW

**Comment by soumith** [comment]
thank you!
